core/java/iesi-core/src/main/java/io/metadew/iesi/openapi/ComponentParser.java

```diff
@@ -0,0 +1,269 @@
+                break;
+            case 2:
+                current.put("response", value);
+                break;
+        }
+        return current;
+    }
+
+
+    public  Component createComponent(HashMap<String, String> partNames, Operation operation,PathItem.HttpMethod operationName, String pathName) {
+        String componentName = buildName(operation.getOperationId(), partNames);
+        String componentType = "http.request";
+        String componentDescription = operation.getDescription();
+        ComponentVersion componentVersion = new ComponentVersion(new ComponentVersionKey(componentName, versionNumber),componentDescription);
+
+        List<ComponentParameter> infos = getInfos(componentName,pathName, operationName);
+        List<ComponentParameter> queryParams = getQueryParams(componentName, operation.getParameters());
+        List<ComponentParameter> headers = getHeaders(componentName, partNames);
+        List<ComponentParameter> params = Stream.of(infos, queryParams, headers).flatMap(Collection::stream).collect(Collectors.toList());
+        return new Component(new ComponentKey(componentName, versionNumber), componentType, componentName, componentDescription, componentVersion, params, new ArrayList<>());
+
+    }
+
+    public  List<ComponentParameter> getInfos(String componentName, String pathName, PathItem.HttpMethod operationName) {
+        List<ComponentParameter> componentParameters = new ArrayList<>();
+        componentParameters.add(new ComponentParameter(new ComponentParameterKey(componentName, versionNumber, "endpoint"), pathName));
+        componentParameters.add(new ComponentParameter(new ComponentParameterKey(componentName, versionNumber, "type"), operationName.name()));
+        componentParameters.add(new ComponentParameter(new ComponentParameterKey(componentName, versionNumber, "connection"), connectionName));
+        return componentParameters;
+    }
+
+    public  List<ComponentParameter> getQueryParams(String componentName, List<Parameter> parameters) {
+        int counter = 1;
+        List<ComponentParameter> queryParams = new ArrayList<>();
+
+        if (parameters != null) {
+            for (Parameter parameter : parameters) {
+                if (parameter.getIn().equals("query")) {
+                    String parameterName = parameter.getName();
+                    queryParams.add(new ComponentParameter(new ComponentParameterKey(componentName, versionNumber, String.format("queryParam.%s", counter)), String.format("%s, #%s#", parameterName, parameterName)));
+                    counter += 1;
+                }
+            }
+        }
+        return queryParams;
+    }
+
+
+    public  List<ComponentParameter> getHeaders(String componentName, HashMap<String, String> partNames) {
+        int position = 0;
+        List<ComponentParameter> parameters = new ArrayList<>();
+        List<String> headerTypes = new ArrayList<>(partNames.keySet());
+
```

Comment:
can't we have header parameters besides the ones in partnames?

Reply:
The only headers that we can have are : Authorization header, Accept Header and Content-Type Header at the moment. So if there are one of these header  the headerTypes will be not empty.

Reply:
those are the only ones we consider when generating the name, right? but in my documentation I can put whatever header in the documentation. So these other headers should also be considered

Reply:
The name generation method should also consider these headers ? We only discussed about security, content-type, and accept header.

Reply:
only the security, content-type, and accept header are mutually exclusive. Other headers are not

Reply:
I see that there are more 50 header that can be add to a request. I have to handle all of them ?

Reply:
only the ones that are present in the documentation being provided. So something like:
```java
List<Parameter> parameters = operation.getParameters();
for (Parameter parameter : parameters) {
   if (parameter.getIn().equals("header")) {
       // check if security, request, response. If not, add
```

Reply:
Got it